app/src/test/java/com/codesoom/assignment/product/service/ProductServiceTest.java

```diff
@@ -0,0 +1,168 @@
+package com.codesoom.assignment.product.service;
+
+import com.codesoom.assignment.common.exception.CatToyNotFoundException;
+import com.codesoom.assignment.product.domain.CatToy;
+import com.codesoom.assignment.product.repository.ProductRepository;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+
+@Nested
+@DisplayName("CatToyService 클래스")
+class ProductServiceTest {
+
+    private ProductService productService;
+    private ProductRepository productRepository;
+
+    private CatToy givenCatToy;
+    private List<CatToy> givenCatToyList = new ArrayList<>();
+    private static final Long EXIST_ID = 1L;
+    private static final Long NOT_EXIST_ID = 100L;
+
+    @BeforeEach
+    public void setUp() {
+        productRepository = mock(ProductRepository.class);
+        productService = new ProductServiceImpl(productRepository);
+
+        setUpFixtures();
+        setUpSaveTask();
+    }
+
+    void setUpFixtures() {
+        givenCatToy = new CatToy(
+                EXIST_ID,
+                "Test Name",
+                "Test Maker",
+                10000,
+                "test Image Url"
+        );
+        givenCatToyList.add(givenCatToy);
+
+        given(productRepository.findAll()).willReturn(givenCatToyList);
+        given(productRepository.findById(EXIST_ID)).willReturn(Optional.of(givenCatToy));
+        given(productRepository.findById(NOT_EXIST_ID)).willReturn(Optional.empty());
+    }
+
+    void setUpSaveTask() {
+        given(productRepository.save(any(CatToy.class))).will(invocation -> {
+            CatToy catToy = invocation.getArgument(0);
+            catToy.setId(2L);
+            givenCatToyList.add(catToy);
+            return catToy;
+        });
+    }
+
+    @Nested
+    @DisplayName("getCatToys 메서드는")
+    class getAllCatToys {
+        @Test
+        @DisplayName("장난감 목록 전체를 반환한다.")
+        void getCatToys() {
+            List<CatToy> catToys = productService.getCatToys();
+
+            Assertions.assertThat(catToys).hasSize(1);
```

Comment:
여기에 `1`이 오면 안 됩니다. 여기에 `1`이 있어야 할 논리적인 근거가 없어요. 이 논리적인 근거는 이 `1`을 써놓은 사람의 머리 속에만 있습니다. `1`의 근거를 코드를 통해 마련해 주어야 합니다. 근거가 마련되면 이 파일에서 `1`은 사라지겠지요.

```suggestion
            Assertions.assertThat(catToys).hasSize( ... );
```

Reply:
그렇네요 반환 받은 리스트의 크기와, 처음에 만든 리스트의 크기가 같다는 것을 테스트하도록 하였습니다.